External/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+set(SKIP_PORTABILITY_TEST ON)
+set(JUST_INSTALL_CEREAL ON)
+
+add_subdirectory(cereal)
+add_subdirectory(glslang)
+set(SPIRV_TOOLS_BUILD_STATIC ON)
+add_subdirectory(SPIRV-Cross)
+add_subdirectory(Vulkan-Headers)
```

Comment:
FetchDependencies is not a problem.  From the man page for mkdir:
-p, --parents
    no error if existing, make parent directories as needed

If a directory exists, it silently is happy and doesn't create any new directory.  The advantage of mkdir -p is that it will create any and all directories down a chain.
Additionally it might be worthwhile to consider using submodules instead of the fetchDependencies script in the future.